sycl/test-e2e/syclcompat/atomic/atomic_class.cpp

```diff
@@ -41,8 +41,8 @@
 #include "../common.hpp"
 #include "atomic_fixt.hpp"
 
-constexpr size_t numBlocks = 64;
-constexpr size_t numThreads = 256;
+constexpr size_t numBlocks = 1;
```

Comment:
What's the reasoning for going to a single block and thread? Surely you want you atomic tests run in parallel? What am I missing?

Reply:
This change does not affect how the atomics are executing at all, since in the test they only ever operate on block local atomic variables.
Here is a simpler version of the test that reproduces the problem:

```
#include <sycl/sycl.hpp>
int main() {
  
  sycl::queue q{sycl::gpu_selector_v};
  
  auto output = sycl::malloc_shared<long>(1, q);
  output[0] = 1;

  q.submit([&](sycl::handler &h) {
     auto ndr =
         sycl::nd_range<1>{sycl::range<1>{64 * 256}, sycl::range<1>{256}};
     h.parallel_for(ndr, [=](sycl::id<1> it) {
       //      h.single_task([=] {

       long c = 0;
       sycl::atomic_ref<long, sycl::memory_order::acq_rel,
                        sycl::memory_scope::device>
           a{c};

       long b = -1;
       output[0] = a.fetch_sub(b);
     });
   }).wait_and_throw();
  std::cout << output[0] << std::endl; //garbage value
}
```
The atomic operations in the test operate on a local variable. The problem is that there is no atomic operation on the output shared usm. I don't think that the above program is well defined. `output` is correct if you either:
- check that the work-group id is e.g. 0, and only writing to output in that case
- use atomics to write to output
- just use a single_task aka block size 1 thread size 1.

Reply:
Right. It does look dodgy, and probably deserves a cleanup in another patch. If I understand you correctly, it doesn't belong in this changeset and should therefore be removed

Reply:
> Right. It does look dodgy, and probably deserves a cleanup in another patch. If I understand you correctly, it doesn't belong in this changeset and should therefore be removed

Yeah I plan to open a new PR for those fixes one I get CI passes here. They are currently not tested at all because seq_cst isn't supported till this PR here.